src/Modules/Settings/Dnn.PersonaBar.Prompt/Services/CommandController.cs

```diff
@@ -16,20 +16,61 @@
 using DotNetNuke.Services.Exceptions;
 using DotNetNuke.Services.Localization;
 using DotNetNuke.Services.Log.EventLog;
+using DotNetNuke.Entities.Portals;
 
 namespace Dnn.PersonaBar.Prompt.Services
 {
     [MenuPermission(MenuName = "Dnn.Prompt")]
     [RequireHost]
-    public class CommandController : ControllerBase
+    public class CommandController : ControllerBase, IServiceRouteMapper
     {
         private static readonly ILog Logger = LoggerSource.Instance.GetLogger(typeof(CommandController));
         private static readonly string[] BlackList = { "smtppassword", "password", "pwd", "pass", "apikey" };
 
-        [HttpGet]
-        public HttpResponseMessage List()
+        private int portalId = -1;
```

Comment:
I suggest naming `portalId` to `_portalId` to differentiate from passed parameter and to follow conventions.